### PR TITLE
vatest: Detect MMU and configure test

### PR DESCRIPTION
**Pull request by harish-24**
1) Check MMU for setting different page sizes 

Patch checks for MMU rather than the previous implementation of check of different architecture. This allows machines with hash page table to run properly irrespective of the sub-arch.

2) Check 16G hugepage as default

This patch checks whether 16G hugepages is default, elsewhere it cannot be configured and used. So check and skip scenarios which are not applicable.

3) Fix pep8 and pylint issues

Signed-off-by: Harish <harish@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by harish-24** [comment]
Job log: [mmu_log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/2132961/mmu_log.txt)

File: memory/vatest.py [hunk 1]
```diff
@@ -44,7 +44,10 @@ def setUp(self):
         self.hsizes = [1024, 2]
         page_chunker = memory.meminfo.Hugepagesize.m
         if distro.detect().arch in ['ppc64', 'ppc64le']:
-            if 'power8' in cpu.get_cpu_arch():
+            mmu_detect = genio.read_file(
+                '/proc/cpuinfo').strip().splitlines()[-1]
+            # Check for "Radix" as this MMU will be explicit in POWER
+            if 'Radix' not in mmu_detect:
```
Comment (sathnaga) [inline memory/vatest.py]: older distros `cpuinfo` would not have `MMU` details, you might have to handle if this test to be run there aswell.

File: memory/vatest.py [hunk 2]
```diff
@@ -68,11 +79,14 @@ def setUp(self):
             for hp_size in self.hsizes:
                 nr_pgs = int((total_mem / 2) / hp_size)
                 genio.write_file(
-                    '/sys/kernel/mm/hugepages/hugepages-%skB/nr_hugepages' % str(hp_size * 1024), str(nr_pgs))
+                    '/sys/kernel/mm/hugepages/hugepages-%skB/'
+                    'nr_hugepages' % str(hp_size * 1024), str(nr_pgs))
             n_pages = genio.read_file(
-                '/sys/kernel/mm/hugepages/hugepages-%skB/nr_hugepages' % str(self.hsizes[0] * 1024)).rstrip("\n")
+                '/sys/kernel/mm/hugepages/hugepages-%skB/nr_huge'
```
Comment (nasastry) [inline memory/vatest.py]: `/sys/kernel/mm/hugepages/hugepages-%skB/nr_hugepages` is used in many of the places ... where ever possible Create an variable and use the same.

Comment (harish-24) [reply memory/vatest.py]: Sure, makes sense

File: memory/vatest.py (hunk 1 above)
Comment (harish-24) [reply memory/vatest.py]: Okay, let me try to handle.

Comment (sathnaga) [reply memory/vatest.py]: @harish-24 looks like we need this below commit, I hope all distros support Radix will have it.
```
commit 3a4c26011da3ed8df1f4f93af82bd5f2b420751f
Author: Aneesh Kumar K.V <aneesh.kumar@linux.vnet.ibm.com>
Date:   Tue Mar 21 22:59:55 2017 +0530

    powerpc/mm: Add translation mode information in /proc/cpuinfo
    
    With this we have on powernv and pseries /proc/cpuinfo reporting
    
    timebase        : 512000000
    platform        : PowerNV
    model           : 8247-22L
    machine         : PowerNV 8247-22L
    firmware        : OPAL
    MMU             : Hash
    
    Reviewed-by: Paul Mackerras <paulus@ozlabs.org>
    Signed-off-by: Aneesh Kumar K.V <aneesh.kumar@linux.vnet.ibm.com>
    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
```

**Pull request merged**